corehq/blobs/migrate.py

```diff
@@ -274,16 +272,96 @@ def should_process(self, doc):
+            try:
+                content = from_db.get(meta.id, bucket)
+            except NotFound:
+                self.not_found += 1
+            else:
+                with content:
+                    self.db.copy_blob(content, meta.info, bucket)
+        return True
+
+    def processing_complete(self, skipped):
+        super(BlobDbBackendExporter, self).processing_complete(skipped)
+        if self.not_found:
+            print("{} of {} blobs were not found in the old blob database. It "
+                  "is possible that some blobs were deleted as part of normal "
+                  "operation during the migration if the migration took a long "
+                  "time. However, it may be cause for concern if a majority of "
+                  "the total number of migrated blobs were not found."
+                  .format(self.not_found, self.total_blobs))
+
+    def should_process(self, doc):
+        return doc.get('domain') == self.domain and doc.get("external_blobs")
+
+
+class SqlFormAttachmentExporter(BaseDocProcessor):
+    def __init__(self, slug, domain):
+        super(SqlFormAttachmentExporter, self).__init__()
+        from corehq.blobs.zipdb import get_blob_db_exporter, ZipBlobDB
+        self.slug = slug
+        self.db = get_blob_db_exporter(self.slug, domain)
+        self.total_blobs = 0
+        self.not_found = 0
+        self.domain = domain
+        if not isinstance(self.db, ZipBlobDB):
+            raise MigrationError(
+                "Expected to find zip blob db backend (got %r)" % self.db)
+
+    def handle_skip(self, doc):
+        return True  # ignore
+
+    def process_doc(self, doc):
+        from corehq.form_processor.backends.sql.dbaccessors import FormAccessorSQL
+        from corehq.blobs import BlobInfo
+        from_db = get_blob_db()
+        for attachment in FormAccessorSQL.get_attachments(doc['_id']):
+            bucket = attachment.blobdb_bucket()
+            blob_id = attachment.blob_id
+            info = BlobInfo(identifier=blob_id, length=attachment.content_length,
+                            digest="md5="+attachment.md5)
+            self.total_blobs += 1
+            try:
+                content = from_db.get(blob_id, bucket)
+            except NotFound:
+                self.not_found += 1
+            else:
+                with content:
+                    self.db.copy_blob(content, info, bucket)
+        return True
+
+    def processing_complete(self, skipped):
+        super(SqlFormAttachmentExporter, self).processing_complete(skipped)
+        if self.not_found:
+            print("{} of {} blobs were not found in the old blob database. It "
+                  "is possible that some blobs were deleted as part of normal "
+                  "operation during the migration if the migration took a long "
+                  "time. However, it may be cause for concern if a majority of "
+                  "the total number of migrated blobs were not found."
```

Comment:
What do you think of moving this string to a global constant since it's used in a number of places (DRY)?